platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/MarkerFileWriter.java

```diff
@@ -0,0 +1,113 @@
+ */
+public class MarkerFileWriter {
+
+    /**
+     * The logger for this class.
+     */
+    private static final Logger LOG = LogManager.getLogger(MarkerFileWriter.class);
+
+    /**
+     * The directory where the marker files are written.  If null, no marker files are written.
+     */
+    private final File markerFileDirectory;
+
+    /**
+     * Creates a new {@link MarkerFileWriter} with the given {@link PlatformContext}.  If the marker file writer is
+     * enabled, files are written to the configured directory.
+     *
+     * @param platformContext the platform context containing configuration.
+     */
+    public MarkerFileWriter(@NonNull final PlatformContext platformContext) {
+        final Path markerFileDirectoryPath = platformContext
+                .getConfiguration()
+                .getConfigData(PathsConfig.class)
+                .getMarkerFilesDir();
+        File directory = null;
+        if (markerFileDirectoryPath != null) {
+            try {
+                Files.createDirectories(markerFileDirectoryPath);
+                directory = markerFileDirectoryPath.toFile();
+            } catch (final IOException e) {
+                LOG.error(
+                        LogMarker.EXCEPTION.getMarker(),
+                        "Failed to create marker file directory: {}",
+                        markerFileDirectoryPath,
+                        e);
+            }
+            if (!Files.isDirectory(markerFileDirectoryPath)) {
+                directory = null;
+                LOG.error(
+                        LogMarker.ERROR.getMarker(),
+                        "failed to create marker file directory, path is already a file: {}",
+                        markerFileDirectoryPath);
+            }
+        }
+        markerFileDirectory = directory;
+    }
+
+    /**
+     * Writes a marker file with the given filename to the configured directory.
+     *
+     * @param filename the name of the marker file to write.
+     */
+    public void writeMarkerFile(@NonNull final String filename) {
+        if (markerFileDirectory == null) {
+            // Configuration did not set a marker file directory.  No need to write marker file.
+            return;
+        }
+        final File markerFile = new File(markerFileDirectory, filename);
+        if (markerFile.exists()) {
+            // No need to create file when it already exists.
+            return;
+        }
+        try {
+            Files.createFile(markerFile.toPath());
+        } catch (final IOException e) {
+            throw new RuntimeException("Failed to create marker file: " + markerFile, e);
+        }
+    }
+
+    /**
+     * Returns the directory where the marker files are written.  May be null.
+     *
+     * @return the directory where the marker files are written.  This may be null.
+     */
+    public Path getMarkerFileDirectory() {
+        if (markerFileDirectory == null) {
+            return null;
```

Comment:
is it worth logging a message to this effect? "Skipping Marker file, directory not set" or something.

Reply:
We might want to be careful with logging. Or at the very least we'd want to rate limit the log. Some of these marker files have the potential to be written many times during error conditions, and an error followed by a gigabyte of log spam is no fun to debug. 🙃

Reply:
good call, I will log an info message.

Reply:
@edward-swirldslabs I think having a static atomic boolean accessed by the constructor is more complexity than it's worth.

Only a couple of these objects are constructed, at boot, so it's not like we're contending with infinite log spam.

IMO I'd not log this at all, but if we *do* want a `null` input parameter to be logged, having 2 of that log message instead of just 1 doesn't make much of a difference

Reply:
I'll log an error if the feature is turned on.